privacyidea/lib/importotp.py

```diff
@@ -533,10 +533,13 @@ def parsePSKCdata(xml_data,
 
         hash_lib = "sha1"
 
-        # Check if hashlib is explicitly set in file
-        if parameters.suite and parameters.suite.string:
-            hash_lib = parameters.suite.string.lower()
-        token["hashlib"] = hash_lib
+        try:
+            # Check if hashlib is explicitly set in file
+            if parameters.suite and parameters.suite.string:
+                hash_lib = parameters.suite.string.lower()
+            token["hashlib"] = hash_lib
+        except Exception as e:
+            log.warning(f"No compatible suite contained, falling back to default {hash_lib}.")
```

Comment:
Just one last comment. If you exit the try block, then ``token["hashlib"]`` will not be set.
And it will default to the server's default (which usually is sha1 ;-)
You might consider, movint the line 540 out of the try-block to line 543 or into a finally (cool):

~~~~python
try:
    if parameters.suite and ...
        hash_lib = ...
except Exception as e:
    log.warning()
finally:
    token["hashlib"] = hash_lib
~~~~

Reply:
Sounds good to me. Remove the exception-catching and move the log.warning to an else-block.